src/toxcore/dht/server/mod.rs

```diff
@@ -746,6 +748,24 @@ impl Server {
         }
     }
 
+    /// Update returned socket address and time of receiving packet
+    fn update_returned_sock(&self, node: &PackedNode, packet_pk: &PublicKey, close_nodes: &mut Kbucket, friends: &mut Vec<DhtFriend>) {
+        if self.pk == node.pk {
```

Comment:
We don't need this `if`. It will always do nothing.

Reply:
A node may have my DHT pk in it's Kbucket. So, my DHT pk may exist in NodesResponse.
Then the node's PK can be updated in my ret_saddr. Is it wrong?